### PR TITLE
Add option for a linear classifer in scANVI

### PR DESCRIPTION
**Pull request by martinkim0**
Thank you for your contribution to scvi-tools. Feel free to modify or omit any
sections below based on your needs.

---

## General

-   [x] I have read the [contributing guide](https://docs.scvi-tools.org/en/stable/contributing/index.html).
-   [x] I have verified that there are no other pull requests open for the same fix or feature.

## Description

#1669 

-   [x] I have linked the relevant issue(s) and/or included a description of the changes.
-   [x] If the changes fully address an issue, I have added `Fixes #<issue number>` to the PR description.

## Documentation

-   [x] I have added appropriate comments and documentation where necessary.
-   [x] I have added a description of the changes to the most recent entry in `docs/release_notes/index.md`.

## Testing

-   [x] I have added relevant tests for my changes.
-   [x] All new and existing tests pass locally.

## Development

-   [x] If the changes are patches for a version, I have added the `on-merge: backport to x.x.x` label.

### CODE REVIEW

**Comment by martinkim0** [comment]
Here is a [benchmark](https://github.com/scverse/scvi-tools/files/11567540/scanvi_linear.pdf) comparing the classifiers on Tabula Muris. Peak validation F1 and accuracy are similar for both classifiers, but the linear classifier is overfitting less in later epochs.

File: scvi/module/_classifier.py [hunk 1]
```diff
@@ -67,3 +67,14 @@ def __init__(
...
+
+class LinearClassifier(nn.Module):
+    """Basic linear classifier."""
+
+    def __init__(self, n_input: int, n_labels: int):
+        super().__init__()
+        self.module = nn.Sequential(nn.Linear(n_input, n_labels), nn.Softmax(dim=-1))
+
+    def forward(self, x):
+        return self.module(x)
```
Comment (adamgayoso) [inline scvi/module/_classifier.py:80]: can we just made n hidden layers = 0  in the default classifier be a linear classifier?

Comment (martinkim0) [reply scvi/module/_classifier.py:80]: the classifier code would need to change slightly for this to work

Comment (adamgayoso) [reply scvi/module/_classifier.py:80]: I think it's worth it just to make the overall api simpler. Also the logits option is important to do things like logsoftmax for numerical stability

**Review by adamgayoso — CHANGES_REQUESTED** [review]

File: scvi/module/_classifier.py [hunk 2]
```diff
@@ -45,20 +45,27 @@ def __init__(
...
-                use_batch_norm=use_batch_norm,
-                use_layer_norm=use_layer_norm,
-                activation_fn=activation_fn,
-                **kwargs,
-            ),
-            nn.Linear(n_hidden, n_labels),
-        ]
+        layers = []
+
+        if n_hidden > 0 and n_layers > 0:
```
Comment (adamgayoso) [inline scvi/module/_classifier.py:50]: can you document this in this class too?

File: tests/models/test_models.py [hunk 3]
```diff
@@ -1112,6 +1112,18 @@ def test_scanvi(save_path):
...
+    adata = synthetic_iid()
+    SCANVI.setup_anndata(
+        adata,
+        "labels",
+        "label_0",
+        batch_key="batch",
+    )
+    model = SCANVI(adata, linear_classifier=True)
+    model.train(max_epochs=1)
+
```
Comment (adamgayoso) [inline tests/models/test_models.py:1125]: let's assert that the classifier nn module has the right structure in this test as well

**Review by adamgayoso — CHANGES_REQUESTED** [review]

File: scvi/module/_classifier.py (hunk 2 above)
Comment (martinkim0) [reply scvi/module/_classifier.py:52]: done

File: tests/models/test_models.py (hunk 3 above)
Comment (martinkim0) [reply tests/models/test_models.py:1131]: done

**Review by adamgayoso — APPROVED** [review]
